src/status_im/contexts/wallet/send/events_test.cljs

```diff
@@ -163,3 +181,907 @@
         result     (dispatch [event-id {:amount amount}])]
     (testing "amount set"
       (is (match? amount (get-in result [:db :wallet :ui :send :amount]))))))
+
+(h/deftest-event :wallet/clean-send-data
+  [event-id dispatch]
+  (let [token-symbol      "ETH"
+        token             {:symbol   "ETH"
+                           :name     "Ether"
+                           :networks #{{:chain-id 421614}
+                                       {:chain-id 11155420}
+                                       {:chain-id 11155111}}}
+        receiver-networks [421614 11155420]]
+    (reset! rf-db/app-db
+      {:wallet {:ui {:send        {:token-display-name token-symbol
+                                   :token              token
+                                   :receiver-networks  receiver-networks}
+                     :other-props :value}}})
+    (is (not (contains? (get-in (dispatch [event-id]) [:db :wallet :ui]) :send)))))
+
+(h/deftest-event :wallet/select-address-tab
+  [event-id dispatch]
+  (let [expected-db {:wallet {:ui {:send {:select-address-tab "tab"}}}}]
+    (reset! rf-db/app-db
+      {:wallet {:ui {:send nil}}})
+    (is (match? expected-db (:db (dispatch [event-id "tab"]))))))
+
+(h/deftest-event :wallet/clean-send-address
+  [event-id dispatch]
+  (reset! rf-db/app-db
+    {:wallet {:ui {:send {:to-address  "0x01"
+                          :recipient   {:recipient-type :saved-address
+                                        :label          "label"}
+                          :other-props :value}}}})
+  (is (not (contains? (get-in (dispatch [event-id]) [:db :wallet :ui :send]) :to-address)))
```

Comment:
We could clean up a few assertions and be more direct, more readable. Possibly in a few other tests you could use the same idea. If we can get away with a simple equality check the test is potentially more robust. For example, the original test still passes even if the event handler changes other parts of the app-db, but this would clearly be a bug given its name/purpose (clean send address).

It could be simply:

```clojure
(h/deftest-event :wallet/clean-send-address
  [event-id dispatch]
  (let [initial-db])
  (reset! rf-db/app-db
    {:wallet {:ui {:send {:to-address  "0x01"
                          :recipient   {:recipient-type :saved-address
                                        :label          "label"}
                          :other-props :value}}}})
  (is (= {:db {:wallet {:ui {:send {:other-props value}}}}}
         (dispatch [event-id]))))
```

In general, I think the event test can catch more and be more useful if we verify the entire output of the handler. The moment we use `get-in` and grab only slices of the map of effects, we are allowing the implementation to change without coverage. Sometimes it makes sense, but here and in some other tests I think it's not beneficial.

Reply:
But you're using `=` in the assertion. Is that ok?

Reply:
FYI: Just opened PR https://github.com/status-im/status-mobile/pull/20825.